operations/login_operation.go

```diff
@@ -0,0 +1,194 @@
+package operations
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+
+	"encoding/json"
+
+	"github.com/degica/barcelona-cli/api"
+	"github.com/degica/barcelona-cli/utils"
+)
+
+// A proxy class to satisfy the interfaces
+type ProxyLoginOperationClient struct {
+	Client *api.Client
+}
+
+func (p ProxyLoginOperationClient) ReloadDefaultClient() (LoginOperationClient, error) {
+	client, error := p.Client.ReloadDefaultClient()
+	return client, error
+}
+
+func (p ProxyLoginOperationClient) LoginWithGithub(endpoint string, token string) (*api.User, error) {
+	return p.Client.LoginWithGithub(endpoint, token)
+}
+
+func (p ProxyLoginOperationClient) LoginWithVault(endpoint string, vault_url string, token string) (*api.User, error) {
+	return p.Client.LoginWithVault(endpoint, vault_url, token)
+}
+
+func (p ProxyLoginOperationClient) Patch(path string, body io.Reader) ([]byte, error) {
+	return p.Client.Patch(path, body)
+}
+
+type LoginOperationClient interface {
+	LoginWithGithub(endpoint string, token string) (*api.User, error)
+	LoginWithVault(endpoint string, vault_url string, token string) (*api.User, error)
+	Patch(path string, body io.Reader) ([]byte, error)
+}
+
+type LoginOperationExternals interface {
+	// User Input Reader
+	Read(secret bool) (string, error)
+
+	// CommandRunner
+	RunCommand(name string, arg ...string) error
+
+	// FileOps
+	FileExists(path string) bool
+	ReadFile(path string) ([]byte, error)
+
+	// Client stufff
+	LoginOperationClient
+	ReloadDefaultClient() (LoginOperationClient, error)
+
+	// Config stuff
+	WriteLogin(auth string, token string, endpoint string) error
+	GetPublicKeyPath() string
+	GetPrivateKeyPath() string
+}
+
+type LoginOperation struct {
+	endpoint    string
+	backend     string
+	gh_token    string
```

Comment:
it is up to you but my golint says locally
``
don't use underscores in Go names; struct field `gh_token` should be `ghToken`
``